### ISSUE TITLE
Support for namespaces, bundles and engines

### ISSUE THREAD

**Issue by mavenastic** [reactjs/react-rails#337]
Hi,

Let's say my Rails app is divided into 2 namespaces, `frontend` and `backend`. By default, the same `components.js` file is loaded when pre-rendering. We can tweak what is loaded at pre-render time through the `server_renderer_options` configuration variable, however it will always use the same content, regardless of the namespace my request comes from.

**Is there a way to create conditionally loaded bundles?**

In order to obtain these bundles, I tried to separate my namespaces into engines. When using `react-rails` in my first engine, I get the following error:

```
ActionView::Template::Error (undefined method `react_component' 
```

**Is there a way to use the helper from inside a Rails engine?**

Thanks in advance!

**Comment by rmosolgo** [comment]
Sorry, somehow I missed this question before!

__Rails engine:__ I haven't tried it, but it seems like someone is using prerender inside an engine: https://github.com/reactjs/react-rails/issues/360. I'd be happy to help you find the problem if you share your view code, controller code & the full stack trace. 

__Conditionally-loaded bundles:__ no, there's no support for this :( React-rails starts up _one_ `ExecJS` context but that's it! In my own app, I have two controller namespaces two, but i've also prefixed the component names so that there aren't naming conflicts.

**Comment by JamesDullaghan** [comment]
We got the helper working in an engine today after a few hours of debugging. Unfortunately in the engine's `ApplicationController`, we are unable to inherit from `ActionController::Base` or we receive the same `undefined method 'react_component'`

To solve this, I inherited from `::ApplicationController` from the dummy app, no longer receive this error.

When we have a bit more time, i'll get the stacktrace up here.
